Assets/Scripts/CameraOutputController.cs

```diff
@@ -95,43 +95,59 @@ public class AsynchronousSocketListener
     public static ManualResetEvent allDone = new ManualResetEvent(false);
     static Socket listener = null;
     static StateObject globalState = null;
-    static bool sending = false;
-
-    public AsynchronousSocketListener()
-    {
-    }
+    static bool sending = false;
 
-    public static void SendFrame(byte[] data)
+
+    public async static void SendFrame(byte[] data)
     {
         if (globalState != null && !sending)
         {
             sending = true;
-
-            string base64Encoded = System.Convert.ToBase64String(data) + "\n";
-
-            byte[] encodedBytes = System.Text.Encoding.ASCII.GetBytes(base64Encoded);
-
 
             Socket socket = globalState.workSocket;
-            Debug.Log("Sending data to TCP socket");
-            // Begin sending the data to the remote device.
-            AsyncCallback callback = new AsyncCallback((IAsyncResult ar) => {
-                try
-                {
-                    int bytesSent = socket.EndSend(ar);
-                    Debug.Log("Sent " + encodedBytes.Length + " bytes to client.");
-                }
-                catch (Exception e)
-                {
-                    Debug.Log("Socket send failed:" + e.ToString());
-                    globalState = null;
-                }
-                sending = false;
-            });
-
-            socket.BeginSend(encodedBytes, 0, encodedBytes.Length, 0, callback, socket);
-        }
-    }
+            //Debug.Log("Sending data to TCP socket");
+
+            try
```

Comment:
psst: Instead of doing all this, you could construct a [`NetworkStream`](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.networkstream?view=netframework-4.8) from the `Socket`, and then write into it using [`BinaryWriter`](https://docs.microsoft.com/en-us/dotnet/api/system.io.binarywriter?view=netframework-4.8). `NetworkStream` has a proper `WriteAsync` method which should be more efficient than a custom implementation.

Reply:
thanks, I'll check that later!